lib/order.rb

```diff
@@ -1,18 +1,111 @@
+require 'csv'
+require 'awesome_print'
+
 module Grocery
   class Order
     attr_reader :id, :products
 
+    @@all_orders = []
+
+# Initialize Method that also calls method
+# to split csv product list string
     def initialize(id, products)
       @id = id
-      @products = products
+      @products = useful_data(products)
     end
-
+# Method for splitting product strings into hashes
+    def useful_data(data)
+      if data.class == String
+        grouped_products = data.split(";")
+        product_pairs = {}
+        grouped_products.each do |grouped_item|
+          separated_item = grouped_item.split(":")
+          product_pairs[separated_item[0]] = separated_item[1].to_f.round(2)
+        end
+        product_pairs
+      elsif data.class == Hash
+        data
+      end
+    end
+# Sum of product cost method
     def total
       # TODO: implement total
+      subtotal = 0
+      @products.each do |product, price|
+        subtotal += price
+      end
+      (subtotal + (subtotal * 0.075)).round(2)
     end
-
+# method to add additional products to an existing order
     def add_product(product_name, product_price)
       # TODO: implement add_product
+      if @products[product_name]
+        false
+      else
+        @products[product_name] = product_price
+        true
+      end
+    end
+# option enhancement: removes product from an existing order
+    def remove_product(product_name)
+      if @products[product_name]
+        @products.delete(product_name)
+        true
+      else
+        false
+      end
     end
+# returns array of all orders in the CSV
+    def self.all
+      @@all_orders = []
```

Comment:
If you're resetting `@@all_orders` each time `self.all` is called, why make it a class variable.  It's just as good as a local variable.